Clear errors/warnings when noEmit is enabled

when I run `build` I get the following output:

```
$ nexus build
1872 ● nexus:plugin:nexus-plugin-prisma Running Prisma generators ...
2916 ● nexus:build Running typegen & extracting types from addToContext calls
 405 ✕ app uncaughtException
       | error  Error: Cannot find module './api/graphql/Admin'
       |        Require stack:
       |        - /Users/cwd/src/server/node_modules/.build/index.js
       |            at Function.Module._resolveFilename (internal/modules/cjs/loader.js:797:15)
       |            at Function.Module._load (internal/modules/cjs/loader.js:690:27)
       |            at Module.require (internal/modules/cjs/loader.js:852:19)
       |            at require (internal/modules/cjs/helpers.js:74:18)
       |            at Object.<anonymous> (/Users/cwd/src/server/node_modules/.build/index.js:23:1)
       |            at Module._compile (internal/modules/cjs/loader.js:959:30)
       |            at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10)
       |            at Module.load (internal/modules/cjs/loader.js:815:32)
       |            at Function.Module._load (internal/modules/cjs/loader.js:727:14)
       |            at Function.Module.runMain (internal/modules/cjs/loader.js:1047:10) {
       |          code: 'MODULE_NOT_FOUND',
       |          requireStack: [ '/Users/cwd/src/server/node_modules/.build/index.js' ]
       |        }
 855 ✕ nexus:build failed to generate artifacts
       | error  Error: 
       |              Nexus artifact generation failed with exit code "1". The following stderr was captured:
       |        
       |                  null
       |            
       |            at Object.generateArtifacts (/Users/cwd/src/server/node_modules/nexus/src/lib/build/artifact-generation.ts:30:11)
       |            at Object.buildNexusApp (/Users/cwd/src/server/node_modules/nexus/src/lib/build/build.ts:85:5)
       |            at Build.parse (/Users/cwd/src/server/node_modules/nexus/src/cli/commands/build.ts:32:5)
error Command failed with exit code 1.
```

I'm in the process of figuring out how my project is different from the generated project. My current thoughts are something about my tsconfig or maybe this is a red herring or some other issue in my code. Regardless the dev server works without issues.

--- a/docs/guides/project-layout.md
+++ b/docs/guides/project-layout.md
@@ -1,8 +1,9 @@
-## Working With `tsconfig.json`
+## Working With tsconfig.json
 
 - Nexus honours settings within `tsconfig.json`.
-- This ensures that Nexus and your IDE perform identical static analysis. If no `tsconfig.json` is present then Nexus will scaffold one for you.
-- Nexus project layout interacts with `tsconfig.json` in the following ways.
+- This ensures that Nexus and your IDE perform identical static analysis.
+- If no `tsconfig.json` is present then Nexus will scaffold one for you.
+- Nexus interacts with `tsconfig.json` in the following ways.
 
 ##### Project Root
 
@@ -22,6 +23,10 @@
 - Out Root is defined by setting `compilerOptions.outDir`.
 - If you do not specify it then Nexus will default to `node_modules/.build`. Unlike with `rootDir` Nexus will not scaffold the default into your `tsconfig.json` because its presence has no impact upon VSCode.
 - You can override its value interactively with `nexus build --out`.
+
+##### Check-Only Builds
+
+if `compilerOptions.noEmit` is set to `true` then Nexus will not output the build. This makes `nexus build` effectively a checker. This option usually [represents user error](https://github.com/graphql-nexus/nexus/issues/702) so by default Nexus will warn when this option is used. In the future ([#800](https://github.com/graphql-nexus/nexus/issues/800)) there will be ways to disable this the warning if it is really your intent.
 
 ## Conventions
 

--- a/src/lib/build/build.ts
+++ b/src/lib/build/build.ts
@@ -89,20 +89,27 @@ export async function buildNexusApp(settings: BuildSettings) {
 
   const runtimePluginManifests = plugins.map(Plugin.entrypointToManifest).filter((pm) => pm.runtime)
 
-  await writeStartModule({
-    layout: layout,
-    startModule: prepareStartModule(
-      tsBuilder,
-      createStartModuleContent({
-        internalStage: 'build',
-        layout: layout,
-        runtimePluginManifests,
-      })
-    ),
-  })
+  if (!layout.tsConfig.content.options.noEmit) {
+    await writeStartModule({
+      layout: layout,
+      startModule: prepareStartModule(
+        tsBuilder,
+        createStartModuleContent({
+          internalStage: 'build',
+          layout: layout,
+          runtimePluginManifests,
+        })
+      ),
+    })
+  }
+
+  const buildOutputLog =
+    layout.tsConfig.content.options.noEmit === true
+      ? 'no emit'
+      : Path.relative(layout.projectRoot, layout.buildOutput)
 
   log.info('success', {
-    buildOutput: Path.relative(layout.projectRoot, layout.buildOutput),
+    buildOutput: buildOutputLog,
     time: Date.now() - startTime,
   })
 

--- a/src/lib/layout/tsconfig.ts
+++ b/src/lib/layout/tsconfig.ts
@@ -87,6 +87,10 @@ export async function readOrScaffoldTsconfig(input: {
     log.warn(`Please set your tsconfig.json include to have "${tscfg.compilerOptions!.rootDir}"`)
   }
 
+  if (tscfg.compilerOptions.noEmit === true) {
+    log.warn('You have set compilerOptions.noEmit in your tsconfig.json. This will prevent `nexus build` from emitting code.')
+  }
+
   // setup out root
 
   if (input.overrides?.outRoot !== undefined) {

--- a/src/lib/tsc.ts
+++ b/src/lib/tsc.ts
@@ -17,12 +17,14 @@ export function createTSProgram(
   layout: Layout,
   options?: ProgramOptions
 ): ts.EmitAndSemanticDiagnosticsBuilderProgram {
-  const compilerCacheOptions = options?.withCache
-    ? {
-        tsBuildInfoFile: getTSIncrementalFilePath(layout),
-        incremental: true,
-      }
-    : {}
+  // Incremental option cannot be set when `noEmit: true`
+  const compilerCacheOptions =
+    options?.withCache && !layout.tsConfig.content.options.noEmit
+      ? {
+          tsBuildInfoFile: getTSIncrementalFilePath(layout),
+          incremental: true,
+        }
+      : {}
 
   log.trace('Create TypeScript program')
 

